homeassistant/components/zwave_js/sensor.py

```diff
@@ -955,17 +970,20 @@ async def async_poll_value(self, _: bool) -> None:
 
     def _get_data_from_statistics(
         self, statistics: ControllerStatisticsDataType | NodeStatisticsDataType
-    ) -> int | None:
+    ) -> Any:
         """Get the data from the statistics dict."""
         if "." not in self.entity_description.key:
-            return cast(int | None, statistics.get(self.entity_description.key))
+            return self.entity_description.convert(
+                statistics.get(self.entity_description.key)
+            )
 
         # If key contains dots, we need to traverse the dict to get to the right value
+        data: Any = statistics
         for key in self.entity_description.key.split("."):
```

Comment:
Please use the `convert` entity description instead to look up the correct item for each sensor instead of doing this general iteration. Pass the description key and the whole statistics to `convert`.